ports/rp2/boards: Add WIZnet-EVB-boards and wiznet6k submodule

wiznet-grace opened the pull request:
### Summary

This PR adds full support for **WIZnet Ethernet controller EVB (Evaluation Board) series**  
(W5100S, W5500, W6100, W6300, W55RP20) to the MicroPython RP2 port.

---

### Key Changes

1. **Migration from wiznet5k → wiznet6k**
   - `W5100S_EVB_PICO`
   - `W5500_EVB_PICO`

2. **New Board Support (7 boards)**
   - `W5100S_EVB_PICO2` – Pico2 with W5100S ([#16280](https://github.com/micropython/micropython/pull/16280))  
   - `W5500_EVB_PICO2` – Pico2 with W5500 ([#16280](https://github.com/micropython/micropython/pull/16280))  
   - `W55RP20_EVB_PICO` – RP2040 + W5500 SiP ([#16476](https://github.com/micropython/micropython/pull/16476))  
     → **Dedicated PIO driver files added**  
   - `W6100_EVB_PICO` – Pico with W6100  
   - `W6100_EVB_PICO2` – Pico2 with W6100  
   - `W6300_EVB_PICO` – Pico with W6300 QSPI  
     → **Dedicated PIO driver files added**  
   - `W6300_EVB_PICO2` – Pico2 with W6300 QSPI  
     → **Dedicated PIO driver files added**

3. **Network Stack Integration**
   - `extmod/network_wiznet6k.c` (~1,197 lines) – full-featured network driver  
   - `lib/wiznet6k` – WIZnet official ioLibrary as a submodule  
   - **LWIP integration** – supported on all boards  
   - **Official HAL support** – stable and validated hardware abstraction  

4. **PIO-based SPI Driver**
   - `wiznet_pio_spi.[c|h|pio]`, `machine_wiznet_pio_spi.c` (~1,223 lines)  
   - High-performance SPI via RP2040/RP2350 PIO  
   - **QSPI support** – W6300 Single/Dual/Quad mode  

---

### Driver Code Size

- `network_wiznet6k.c`: 1,197 lines  
- PIO-related files: 1,223 lines  
- **Total ~2,420 lines added**

---

### File Structure

> extmod/
> └── network_wiznet6k.c
> 
> lib/
> └── wiznet6k/ # new submodule
> 
> ports/rp2/
> ├── machine_wiznet_pio_spi.c
> ├── wiznet_pio_spi.[c|h|pio]
> └── boards/
> ├──── W5100S_EVB_PICO2/
> ├──── W5500_EVB_PICO2/
> ├──── W55RP20_EVB_PICO/ # PIO-based SPI
> ├──── W6100_EVB_PICO/
> ├──── W6100_EVB_PICO2/
> ├──── W6300_EVB_PICO/ # PIO-based SPI + QSPI
> └──── W6300_EVB_PICO2/ # PIO-based SPI + QSPI


---

### Testing

- Verified **successful builds** on all supported WIZnet EVB boards  
- Verified **TCP (loopback, client/server) and DHCP functionality** on all WIZnet boards  
- Verified **QSPI Single/Dual/Quad modes** on W6300 boards  

---

### Trade-offs and Alternatives

- Driver code size increases (**~2,420 LOC**) due to the new driver and PIO/library integration  
- In return, this provides a **unified driver model** across all WIZnet EVB boards with official ioLibrary integration, plus access to TOE and QSPI features  
- Alternative: keep lwIP-only raw mode to reduce size, but this would forgo TOE and QSPI optimisations

MilhouseVH commented on ports/rp2/boards/W5100S_EVB_PICO2/readme.md:
````diff
@@ -0,0 +1,15 @@
+
+```
+>>> import network
+>>> nic = network.WIZNET5K()
````
```suggestion
>>> nic = network.WIZNET6K()
```
... and presumably the same in the other W5K examples below?

An example demonstrating the use of an explicit call to SPI/PIO_SPI might be nice too, as that allows the default speeds to be overridden... last time I tried this the SPI and PIO_SPI constructor calls weren't 100% compatible.

wiznet-grace replied:
Thank you for pointing this out  
You are right — I missed updating the README. I’ll fix this and add the correct README sections for each board.
I’ll also update the examples to use `WIZNET6K()` and add explicit `SPI/PIO-SPI` initialization examples as you suggested.
Thanks again for the helpful feedback!

SteveSpencer commented:
See #18024. The existing wiznet5k code does not propagate the hostname when using DHCP.
I have a tentative fix, but wondered if this was already resolved in the new wiznet6k code?
Would it be possible to include it?
if not, otherwise I would need to do a PR for the 5k code and then potentially another one for the 6k code.

I also identified that when the hostname is set by the DHCP server (such as dnsmasq), it is ignored by LWIP (so not just the wiznet drivers), but it is possible to workaround that by sending a DNS request for the assigned IP to get the hostname, and then set it.

wiznet-grace commented:
> [#18024를](https://github.com/micropython/micropython/issues/18024) 참조하세요 . 기존 wiznet5k 코드는 DHCP를 사용할 때 호스트 이름을 전파하지 않습니다. 임시 해결책이 있는데, 새 wiznet6k 코드에서 이 문제가 이미 해결되었는지 궁금합니다. 이 문제를 해결할 수 있을까요? 새 wiznet6k 코드에 포함할 수 있을까요? 만약 없다면 5k 코드에 대한 PR을 먼저 작성하고, 6k 코드에 대한 PR을 다시 작성해야 할 수도 있습니다.
> 
> 또한 DHCP 서버(예: dnsmasq)에서 호스트 이름을 설정하면 LWIP에서 무시된다는 사실도 확인했습니다(따라서 wiznet 드라이버뿐만 아니라). 하지만 할당된 IP에 DNS 요청을 보내 호스트 이름을 얻은 다음 설정하면 이 문제를 해결할 수 있습니다.

wiznet-grace commented:
> See #18024. The existing wiznet5k code does not propagate the hostname when using DHCP. I have a tentative fix, but wondered if this was already resolved in the new wiznet6k code? Would it be possible to include it? if not, otherwise I would need to do a PR for the 5k code and then potentially another one for the 6k code.
> 
> I also identified that when the hostname is set by the DHCP server (such as dnsmasq), it is ignored by LWIP (so not just the wiznet drivers), but it is possible to workaround that by sending a DNS request for the assigned IP to get the hostname, and then set it.

Thank you for pointing this out! I’ve revised the code accordingly and will push the changes again. To summarize the modifications:


- **MICROPY_PY_LWIP = 1 (LwIP stack enabled)**

In this mode, the DHCP hostname is taken from MicroPython’s modnetwork layer. By default it uses the value defined in the board configuration (mpconfigboard.h via MICROPY_PY_NETWORK_HOSTNAME_DEFAULT), and it can also be overridden at runtime in Python with network.hostname("...").
![lwip_hostname](https://github.com/user-attachments/assets/98afec18-a225-45bc-ac6b-5e0af9a0569b)



- **MICROPY_PY_LWIP = 0 (ioLibrary TOE stack)**

In this mode, the DHCP client logic comes from ioLibrary. Here, the hostname is hardcoded as "WIZnet" with the last 3 bytes of the MAC address appended, resulting in something like WIZnetABC123. As seen in Wireshark captures, DHCP Option 12 is always set in this format.
![toe_hostname](https://github.com/user-attachments/assets/6bd6b188-da5a-408c-a89f-62ef4148c57f)

In summary:

- LwIP mode: uses the hostname defined in the board configuration (mpconfigboard.h).
- TOE/ioLibrary mode: hostname is always fixed to WIZnet+MAC suffix.

Thanks again for highlighting this—it helped clarify the difference between the two paths.

jkozniewski commented:
@MilhouseVH regarding the `fatal error -5` - I got rid of it doing this on start (v 1.26.1 on W5500-EVB-PICO-POE):
```python
 # Ensure clean state
    nic.active(False)
    time.sleep(0.5)
    nic.active(True)
    time.sleep(0.5)
```
So it probably means that the W5500 chip state is not reset on soft reboot (similarly to PIO code if I remeber correctly and few peripherals as well...)

MilhouseVH commented on ports/rp2/boards/W55RP20_EVB_PICO/readme.md:
````diff
@@ -0,0 +1,22 @@
+```
+>>> import network
+>>> from machine import Pin, WIZNET_PIO_SPI
+>>> spi = WIZNET_PIO_SPI(baudrate=10_000_000, sck=Pin(18), cs=Pin(17), mosi=Pin(19), miso=Pin(16))
````
Are you sure about these pins for the W55RP20? These are the pins for the W5500 (PICO/PICO2). I think for the W55RP20 the pins should be:

```suggestion
>>> spi = WIZNET_PIO_SPI(baudrate=10_000_000, sck=Pin(21), cs=Pin(20), mosi=Pin(23), miso=Pin(22))
```

MilhouseVH reviewed (changes requested):
Just trying to develop a minimalist example that reproduces a UART "stall" issue when using the WIZNET6K library with PIO_SPI (regular SPI seems OK), and noticed an issue with your W55RP20 example code.

MilhouseVH commented:
Sample program now attached. This should work automatically on a W5500 with/without `WIZNET_PIO_SPI` support.

If the `WIZNET_PIO_SPI` module is enabled in the build but is NOT imported then the `UART` behaves as expected and all data is output over the `UART` correctly.

However when the `WIZNET_PIO_SPI` module IS loaded then nothing more is output over the `UART` after the `WIZNET_PIO_SPI` object is created (ie. the `log_print()` at line 35 does not appear on the `UART`) but the program does complete successfully as the data is output to the REPL (if using Thonny etc.) - it's just the `UART` that is "stalled".

```
from machine import Pin, UART
from network import WIZNET6K as WIZNET
try:
    from machine import WIZNET_PIO_SPI
except:
    from machine import SPI
import time
import os

BOARD_IS_W5500 = os.uname().machine.startswith('W5500')

uart = UART(0, 115_200, parity=None, stop=1, bits=8, tx=Pin(0), rx=Pin(1), timeout=250, timeout_char=250)

def ymdhms():
    now = time.localtime()
    return f'{now[0]:04d}-{now[1]:02d}-{now[2]:02d} {now[3]:02d}:{now[4]:02d}:{now[5]:02d}'

def log_print(msg):
    print(msg)
    uart.write(f'\r[{ymdhms()}] {msg}\r\n')

def connect():
    log_print('Connecting...')
    time.sleep(1)

    if BOARD_IS_W5500:
        wiz_pins = {'sck': 18, 'mosi': 19, 'miso': 16, 'cs': 17, 'rst': 20, 'int': 21} #W5500_EVB_PICO, W5500_EVB_PICO2
    else:
        wiz_pins = {'sck': 21, 'mosi': 23, 'miso': 22, 'cs': 20, 'rst': 25, 'int': 24} #W55RP2_EVB_PICO etc.

    try:
        spi = WIZNET_PIO_SPI(baudrate=10_000_000, sck=Pin(wiz_pins['sck']), cs=Pin(wiz_pins['cs']), mosi=Pin(wiz_pins['mosi']), miso=Pin(wiz_pins['miso']))
    except:
        spi = SPI(baudrate=10_000_000, sck=Pin(wiz_pins['sck']), mosi=Pin(wiz_pins['mosi']), miso=Pin(wiz_pins['miso']))
    log_print(f'SPI: {spi}')
    time.sleep(1)

    nic = WIZNET(spi, wiz_pins['cs'], wiz_pins['rst'])  # spi, cs, reset pin
    log_print(f'NIC: {nic}')

    # HACK: Disable then enable NIC to silence "fatal error -5", see #17242
    nic.active(False)
    nic.active(True)

    nic.ifconfig('dhcp')

    start = time.ticks_ms()
    while not nic.isconnected():
        time.sleep_ms(50)

    ifconfig = nic.ifconfig()
    log_print(f"IP : ip '{ifconfig[0]}', mask '{ifconfig[1]}', gateway '{ifconfig[2]}', dns '{ifconfig[3]}'")

    return

connect()

log_print('END!')
```

MilhouseVH commented:
The expected output when using `machine.SPI` on W5500-EVB-PICO2 (built with `WIZNET_PIO_SPI` module):

<img width="1165" height="866" alt="image" src="https://github.com/user-attachments/assets/19ae2ed0-8f79-41b9-85f3-29c09b06557c" />

The actual output, same firmware but now using the `machine.WIZNET_PIO_SPI` instead of `machine.SPI`:

<img width="1155" height="834" alt="image" src="https://github.com/user-attachments/assets/46aa71b7-017c-49df-a323-ebaf0df6f3e6" />

Hard reset, soft reset makes no difference - when using `WIZNET_PIO_SPI` the `UART` is dead just as soon as the `WIZNET_PIO_SPI` object is created.

MilhouseVH commented:
Note that I'm testing with the W5500-EVB-PICO2 firmware that I've built with WIZNET_PIO_SPI support so that I can easily switch between WIZNET_PIO_SPI and SPI for testing...

The stock W55RP20-EVB-PICO firmware (that only supports WIZNET_PIO_SPI with WIZNET6K) also clobbers the UART in exactly the same way as the W5500-EVB-PICO2 (when using WIZNET_PIO_SPI) so I think it's a valid test even if the W5500-EVB-PICO2 firmware isn't entirely "stock".

I've attached my custom W5500-EVB-PICO2 firmware with WIZNET_PIO_SPI support, [here](https://github.com/user-attachments/files/24357569/W5500_EVB_PICO2-with-WIZNET_PIO_SPI.zip)

wiznet-grace committed:
rp2/boards: Add WIZnet EVB boards and wiznet6k submodule.

Signed-off-by: wiznet-grace <Grace@wiznet.io>

MilhouseVH commented:
In fact, the W5500-EVB-PICO/W5500-EVB-PICO2 `readme.md` examples succeed/fail in the same way:

W5500-EVB-PICO (RP2040) ✅ :
<img width="758" height="316" alt="image" src="https://github.com/user-attachments/assets/f6366199-9e65-4aa0-b8fb-7be688f76566" />

W5500-EVB-PICO2 (RP2350) ❌:
<img width="769" height="197" alt="image" src="https://github.com/user-attachments/assets/f0da100e-e2b0-4201-980f-006adeac95f1" />

The `fatal error -5` can be avoided by an initial `nic.active(False)`:
<img width="764" height="314" alt="image" src="https://github.com/user-attachments/assets/a1c0e27b-84dc-4d5b-811c-84670e058d4f" />

MilhouseVH commented:
@wiznet-grace Good news - it looks like I did somehow damage my W5500-EVB-PICO2, it's toast (as far as SPI is concerned!) I suspect I may have plugged it into a header incorrectly and damaged one or more of the pins. 😢 

The replacement W5500-EVB-PICO2 board arrived earlier today and is connecting fine with WIZNET6K+SPI (using 1.27 WIZnet firmware) - apologies for the false alarm!

Also, I have now tested W6100-EVB-PICO2 (SPI) and W6300-EVB-PICO2 (WIZNET_PIO_SPI), and they're working fine too, so I have no further issues with this PR. 👍

wiznet-grace commented on ports/rp2/boards/W55RP20_EVB_PICO/mpconfigboard.cmake:
```diff
@@ -0,0 +1,8 @@
+# cmake file for Wiznet W55RP20-EVB-Pico.
+set(PICO_BOARD wiznet_w5100s_evb_pico)
```
Good catch! This follows the existing convention from commit c364301817:

All WIZnet Pico boards (RP2040-based) share the same hardware layout and pinout, so they use the `wiznet_w5100s_evb_pico` base configuration. Pico2 boards use `pico2` for RP2350.

Using board-specific `PICO_BOARD` names would require adding them to pico-sdk first, which hasn't been done yet for W55RP20, W6100, and W6300.

Also, thank you so much for testing these boards! I really appreciate your thorough review.

wiznet-grace commented:
Hi @dpgeorge! I noticed that #17447 has been moved to v1.28.0.

This PR (#18035) adds support for WIZnet Pico2 boards and new chip variants (W6100, W6300, W55RP20), and migrates from wiznet5k to wiznet6k (the unified library for both 5k and 6k series chips).

As @MilhouseVH mentioned, it might make sense to land this PR first, since it provides the foundation for the newer chips that #17447 would need to support.

Would appreciate your review when possible.

Thank you for maintaining MicroPython!

tofDE commented:
Hi,
I have been using this version (MicroPython v1.27.0-WIZnet on 2025-12-30; W5500-EVB-Pico2 with RP2350) on a pico2 with an external W5500 based LAN Adapter and it has been working quite well.
that being said, I would have a question regarding the flash size?
Is there any reason to have the same Flash available on the pico2 as on the pico (about 1.4MB).
This is fine for a pico having 2MB in total. 
For a pico2 with twice the flash, I would have expected to have about 3MB for files, leaving the last MB for Micropython itself...?

regards
tof

wiznet-grace commented:
Hi @dpgeorge,

I'd like to add two additional commits to this PR:

1. **[[extmod/network_wiznet5k,6k: implement timeout and setsockopt for provided stack.](https://github.com/WIZnet-ioNIC/WIZnet-EVB-Pico-micropython/commit/2906a93f4227caeec440b9a36bf90f4c2676583b)](https://github.com/WIZnet-ioNIC/WIZnet-EVB-Pico-micropython/commit/2906a93f4227caeec440b9a36bf90f4c2676583b)**
   - Implements `settimeout()` and `setsockopt()` (SO_REUSEADDR, SO_BROADCAST, SO_KEEPALIVE, SO_SNDTIMEO/SO_RCVTIMEO) which were previously TODO stubs returning EINVAL.

2. **[[extmod/network_wiznet5k,6k: Return 0 (EOF) on recv when peer closes connection.](https://github.com/WIZnet-ioNIC/WIZnet-EVB-Pico-micropython/commit/5a8e2932390c5238c69cb317bb792d0dbde22c2e)](https://github.com/WIZnet-ioNIC/WIZnet-EVB-Pico-micropython/commit/5a8e2932390c5238c69cb317bb792d0dbde22c2e)**
   - When `SOCK_CLOSE_WAIT` is detected with no remaining RX data, `recv()` now returns 0 (EOF) instead of `ENOTCONN`, matching POSIX behavior. This was a pre-existing issue that caused failures in HTTP client usage.

Would it be okay to include these in this PR? Happy to split into a separate PR if you'd prefer.

wiznet-grace commented:
Rebased onto latest master (b05f9a5). Resolved conflicts in `.gitmodules`, `W5100S_EVB_PICO/mpconfigboard.cmake`, and `W5500_EVB_PICO/mpconfigboard.cmake` — keeping `WIZNET6K` and compiler flags while incorporating the upstream `MICROPY_HW_FLASH_STORAGE_BYTES` cmake migration.

Also aligned all 9 WIZnet EVB boards to consistently set `MICROPY_PY_NETWORK`, `MICROPY_WIZNET_PIO`, and `MICROPY_HW_FLASH_STORAGE_BYTES` in cmake (moved from `.h`).

MilhouseVH commented:
b05f9a5bee6146dd33b0675ae7d5d9120915600c tested fine on top of current master (d901e9834939372f68974010f32e146596a69bb0): W55RP20-EVB-PICO, W5500-EVB-PICO, W5500-EVB-PICO2 and W6300-EVB-PICO2. 👍

wiznet-grace commented:
Rebased onto master and force-pushed with the fixes amended in.

This addresses a W6300-EVB-Pico2 report where `status()` was stuck at 0 with the link LEDs on, DHCP timing out, and `wiznet6k_send_ethernet: fatal error -5` on transmit. All changes are in `extmod/network_wiznet6k.c`:

- `wizphy_getphylink()` reads the latching-low BMSR link bit over MDIO only once, so it reported link-down after any reset. `isconnected()` and `status()` now read `PHYSR` directly on W6100/W6300.
- A transfer error took the netif down with nothing to bring it back up, which also stopped RX polling and the lwIP timers. Those calls are removed.
- `wiznet6k_deinit()` ran after the chip reset, so lwIP's multicast leave was sent on a closed socket and printed a spurious `-5`. It now runs before the reset.
- `regs()` passed a raw offset where W6100/W6300 expect `(offset << 8) | block`, so the dump was meaningless. Fixed — it now reaches `Sn_SR` at 0x30.
- `DHCP_time_handler()` had no caller, so `DHCP_run()` never retransmitted a lost request. The DHCP loop now calls it once a second.
- `wiznet6k_init()` pushed a zeroed `netinfo` before the defaults were filled in, clearing SHAR and the IP config. The premature call is removed.

Basic ethernet operation confirmed on all boards.

MilhouseVH commented:
Thanks @wiznet-grace 

It took me a while to work out this PR (4671b80d1db06c69cac5d669aa2a1e5db53773f9) doesn't apply cleanly on current master (b006887db6e4b18bce967ea87c7e63398f60b296) but does apply clean on the older commit 06bcfd5b74c6d275ae0991a19dab8704299e4e05 from 1 August 2026 - dunno what the problem is, but this PR may require another small rebase?

I've built this PR for all the supported boards, no issues.

I've tested it on `W5500_EVB_PICO`, `W5500_EVB_PICO2`, `W55RP20_EVB_PICO` and `W6300_EVB_PICO2` - again no issues, all working as expected.

Unfortunately I don't have the time to review the source code in detail.

@dpgeorge is there any chance of this landing soon? I'm sure that would make a lot of people very happy! Thanks. 😃

wiznet-grace commented:
> Thanks @wiznet-grace
> 
> It took me a while to work out this PR ([4671b80](https://github.com/micropython/micropython/commit/4671b80d1db06c69cac5d669aa2a1e5db53773f9)) doesn't apply cleanly on current master ([b006887](https://github.com/micropython/micropython/commit/b006887db6e4b18bce967ea87c7e63398f60b296)) but does apply clean on the older commit [06bcfd5](https://github.com/micropython/micropython/commit/06bcfd5b74c6d275ae0991a19dab8704299e4e05) from 1 August 2026 - dunno what the problem is, but this PR may require another small rebase?
> 
> I've built this PR for all the supported boards, no issues.
> 
> I've tested it on `W5500_EVB_PICO`, `W5500_EVB_PICO2`, `W55RP20_EVB_PICO` and `W6300_EVB_PICO2` - again no issues, all working as expected.
> 
> Unfortunately I don't have the time to review the source code in detail.
> 
> @dpgeorge is there any chance of this landing soon? I'm sure that would make a lot of people very happy! Thanks. 😃

Rebased onto current master (b006887) — the rebase is the only change, no code
changes since you tested.

Thank you for the builds and the hardware testing, it's always appreciated.

MilhouseVH commented:
I really hope this finally lands in 1.30... it's missed the cut enough times to perhaps start thinking there's some sort of agenda against it. \o/

projectgus commented on extmod/modmachine.h:
```diff
@@ -196,6 +196,9 @@ typedef struct _mp_machine_soft_spi_obj_t {
 } mp_machine_soft_spi_obj_t;
 
+// PIO QSPI object.
+typedef mp_machine_soft_spi_obj_t machine_wiznet_pio_spi_obj_t;
```
`extmod/` is generally for cross-port machine features. If I understand correctly, the "Wiznet PIO SPI" function is only relevant for rp2 port.

Is there anything Wiznet-specific about this class? For MicroPython, it seems like it would make more sense to:

1. Add a `machine.PIOSPI` object to the rp2 port, that can be used anywhere that regular `machine.SPI` can be used.
2. Add support to the Wiznet driver to accept any "SPI object" as the driver, including the "PIOSPI" object.

What do you think?

I'm not necessarily saying you have to make this change for this PR, but if we're adding these features then it's important for us to think about how it works for the whole of MicroPython and not just the Wiznet drivers.

wiznet-grace replied:
Agreed. I'll rework it as a general-purpose PIO SPI rather than a WIZnet-specific one, and submit that as a separate PR.

projectgus commented on extmod/network_wiznet6k.c:
```diff
@@ -0,0 +1,1239 @@
+    // Configure wiznet for raw ethernet frame usage.
+
+    // Configure buffers for fast MACRAW
+    #if _WIZCHIP_ < W5200
+    uint8_t sn_size[8] = {8, 0, 0, 0, 8, 0, 0, 0};
+    #elif (_WIZCHIP_ == W5200 || _WIZCHIP_ == W5300 || _WIZCHIP_ == W5500 || _WIZCHIP_ == W6100)
+    uint8_t sn_size[16] = {16, 0, 0, 0, 0, 0, 0, 0, 16, 0, 0, 0, 0, 0, 0, 0};
```
We already have drivers for the Wiznet 5K series in `network_wiznet5k.c`. How should we approach this, can we remove the `network_wiznet5k.c` driver and replace it with this one?

wiznet-grace replied:
Answered in the main review thread — yes, it can be removed and replaced with this one.

projectgus commented on ports/rp2/boards/W5100S_EVB_PICO/mpconfigboard.cmake:
```diff
@@ -1,9 +1,13 @@
 if(NOT DEFINED MICROPY_HW_FLASH_STORAGE_BYTES)
     set(MICROPY_HW_FLASH_STORAGE_BYTES 1441792)  # 1408 * 1024
 endif()
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-but-set-variable -Wno-unused-variable -Wno-misleading-indentation -Wno-incompatible-pointer-types -Wno-error=unused-function -Wno-comment -Wno-unused-function")
```
We can't disable warnings for all source files like this.

If there are external files which need this change applied then can do it for single source files in `ports/rp2/CMakeLists.txt` like this:
https://github.com/micropython/micropython/blob/master/ports/rp2/CMakeLists.txt#L641

However if the warnings are coming from MicroPython code then they need to be fixed, not ignored.

projectgus replied:
(Same comment applies for all of the new mpconfigboard.cmake files.)

wiznet-grace replied:
Yes, I added those deliberately, because the vendored ioLibrary_Driver still has warnings of its own. I'll take them out and check whether any of the warnings are coming from the MicroPython code rather than the library.

projectgus commented on ports/rp2/boards/W5100S_EVB_PICO/mpconfigboard.cmake:
```diff
@@ -1,9 +1,13 @@
     set(MICROPY_HW_FLASH_STORAGE_BYTES 1441792)  # 1408 * 1024
 endif()
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-but-set-variable -Wno-unused-variable -Wno-misleading-indentation -Wno-incompatible-pointer-types -Wno-error=unused-function -Wno-comment -Wno-unused-function")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mthumb -mcpu=cortex-m0plus")
```
This line shouldn't be necessary, the rp2 build process should already be setting these flags.

projectgus replied:
(Same comment applies for all of the new mpconfigboard.cmake files.)

wiznet-grace replied:
Understood — I'll test that and confirm.

projectgus commented on ports/rp2/boards/W5100S_EVB_PICO/readme.md:
````diff
@@ -6,10 +6,13 @@ To use network / socket based code, connect ethernet port to network with DHCP r
 ```
 >>> import network
->>> nic = network.WIZNET5K()
+>>> nic = network.WIZNET6K()
````
This will be a breaking change for anyone already using this board, the driver name is changing.

It's also a confusing API if some boards need `WIZNET5K()` for Wiznet 5K and some need `WIZNET6K()`. See earlier comment about the future of the wiznet5k driver.

wiznet-grace replied:
Answered in the main review thread!

wiznet-grace commented:
@projectgus Thanks for the review, and no need to apologise for the wait.

First, some background that should make the rest clearer.
The wiznet6k submodule is the [ioLibrary_Driver](https://github.com/Wiznet/ioLibrary_Driver) that WIZnet itself uses, brought in as-is.
That library was unified last year so the 5k and 6k chips are supported by one codebase.
Sorry for the confusion the wiznet6k name caused — my intent was for it to replace the wiznet5k submodule.
The name itself doesn't matter to me; what I care about is that ioLibrary_Driver goes in unchanged.

Splitting the PR up makes sense, and I'll do it in the order you suggested.

1. On the future of the wiznet5k driver: network_wiznet5k.c can be removed and replaced with this one. I'd keep network.WIZNET5K() as the public name on every board and swap the implementation behind it, so existing code keeps working and there's a single class name across all the chips. There are also parts that were never implemented in the 5k driver, such as socket [timeout](https://github.com/WIZnet-ioNIC/WIZnet-EVB-Pico-micropython/tree/fix-wiznet-timeout), which I already have working — I'll submit that separately.

2. On documentation: agreed, I'll add it for the new driver.

3. On other ports: the network_wiznet6k driver is only wired up for rp2 at the moment, so that's the only port it's been tested on.
